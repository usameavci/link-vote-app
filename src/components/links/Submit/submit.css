.link-submit {
    display: grid;
    padding: 0.5rem;
    grid-gap: 1rem;
    border-radius: 4px;
    align-items: center;
    grid-auto-flow: column;
    background-color: #f7f7f7;
    border: 1px solid #eaeaea;
    justify-content: flex-start;
}

.link-submit-icon {
    width: 64px;
    height: 64px;
    display: flex;
    font-size: 2rem;
    border-radius: 4px;
    align-items: center;
    justify-content: center;
    background-color: #ececec;
    border: 1px solid #c1c1c1;
}

.link-submit-title {
    font-weight: 700;
    font-size: 1.75rem;
}
