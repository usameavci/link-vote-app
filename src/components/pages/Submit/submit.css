.page-submit {
}

.page-submit-back {
  margin: 1rem 0;
  display: block;
  font-weight: 700;
  font-size: 1.25rem;
}

.page-submit-back .fa {
  margin-right: 1rem;
}

.page-submit-title {
  margin: 3rem 0;
}

.page-submit-form {
}

.page-submit-group {
  display: grid;
  grid-gap: 0.5rem;
  margin-bottom: 2rem;
}

.page-submit-label {
  margin-left: 0.5rem;
}

.page-submit-input {
  padding: 1rem;
  font-size: 1.1rem;
  border-radius: 4px;
  border: 1px solid #555;
}

.page-submit-button {
  border: 0;
  color: #fff;
  padding: 1rem;
  min-width: 150px;
  font-weight: 700;
  font-size: 1.25rem;
  border-radius: 32px;
  background-color: #000;
}
