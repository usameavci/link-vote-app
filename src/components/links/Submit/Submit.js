import React from "react";
import { Link } from "react-router-dom";

import "./submit.css";

const Submit = () => {
  return (
    <Link data-testid="link-submit" className="link-submit" to="/submit">
      <div className="link-submit-icon">
        <i className="fa fa-plus" />
      </div>
      <div className="link-submit-title">SUBMİT A LİNK</div>
    </Link>
  );
};

export default Submit;
