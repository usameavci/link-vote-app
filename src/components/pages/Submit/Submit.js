import upperCase from "lodash/upperCase";
import { Link } from "react-router-dom";
import React, { useContext, useState } from "react";

import Notification from "../../../utils/Notification";
import { LinkContext } from "../../../contexts/LinkContext";

import "./submit.css";

const PageSubmit = () => {
  const [url, setUrl] = useState("");
  const [title, setTitle] = useState("");
  const { addLink } = useContext(LinkContext);

  const handleFormSubmit = (e) => {
    if (!url || !title) {
      Notification.error("Title and url fields can not be empty!");

      return e.preventDefault();
    }

    addLink({ url, title }).then(() => {
      Notification.success(`<b>${upperCase(title)}</b> added`);

      setUrl("");
      setTitle("");
    });

    return e.preventDefault();
  };

  return (
    <div data-testid="page-submit" className="page-submit">
      <Link to="/" className="page-submit-back">
        <i className="fa fa-arrow-left" />
        <span>Return to list</span>
      </Link>

      <h1 className="page-submit-title">Add New Link</h1>

      <form data-testid="submit-form" className="page-submit-form" onSubmit={handleFormSubmit}>
        <div className="page-submit-group">
          <label className="page-submit-label" htmlFor="title">
            Link Name:
          </label>
          <input
            data-testid="submit-form-input-title"
            type="text"
            value={title}
            placeholder="e.g. Alphabet"
            className="page-submit-input"
            onChange={(e) => setTitle(e.target.value)}
          />
        </div>

        <div className="page-submit-group">
          <label className="page-submit-label" htmlFor="title">
            Link URL:
          </label>
          <input
            data-testid="submit-form-input-url"
            type="text"
            value={url}
            className="page-submit-input"
            placeholder="e.g. http://abc.xyz"
            onChange={(e) => setUrl(e.target.value)}
          />
        </div>

        <div className="text-right">
          <button data-testid="submit-form-button-submit" type="submit" className="page-submit-button">
            ADD
          </button>
        </div>
      </form>
    </div>
  );
};

export default PageSubmit;
