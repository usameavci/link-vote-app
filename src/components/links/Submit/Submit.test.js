import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import Submit from "./Submit";

describe("Links sort component tests", () => {
  it("renders without crashing", () => {
    const component = (
      <Router>
        <Submit />
      </Router>
    );

    const { getByTestId } = render(component);

    expect(getByTestId("link-submit")).toBeInTheDocument();
  });

  it("matches snapshots", () => {
    const component = (
      <Router>
        <Submit />
      </Router>
    );

    const tree = renderer.create(component);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
