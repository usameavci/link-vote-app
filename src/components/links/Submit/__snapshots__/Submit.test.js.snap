// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links sort component tests matches snapshots 1`] = `
<a
  className="link-submit"
  data-testid="link-submit"
  href="/submit"
  onClick={[Function]}
>
  <div
    className="link-submit-icon"
  >
    <i
      className="fa fa-plus"
    />
  </div>
  <div
    className="link-submit-title"
  >
    SUBMİT A LİNK
  </div>
</a>
`;
